cmd/rtrmon/rtrmon.go

```diff
@@ -382,14 +382,13 @@ func (c *Client) HandlePDU(cs *rtr.ClientSession, pdu rtr.PDU) {
 
 		key := fmt.Sprintf("%s-%d-%d", pdu.Prefix.String(), pdu.MaxLen, pdu.ASN)
 		c.compRtrLock.Lock()
+		defer c.compRtrLock.Unlock()
 
 		if pdu.Flags == rtr.FLAG_ADDED {
 			c.vrpsRtr[key] = &vrp
 		} else {
 			delete(c.vrpsRtr, key)
 		}
-
-		c.compRtrLock.Unlock()
```

Comment:
I'm not 100% sure here. I need to test this out.  I don't know exactly when a defer is run inside a switch statement. I've never deferred inside one before. 

Give me a few

Reply:
https://play.golang.org/p/L99fOS46XwO

So as I though, defer is only called at the exit of the function, not the 'exit' of a case statement. It's probably still good in this context as there is nothing else happening besides the case. i.e. there is nothing else being executed after the switch statement. However if something were to be added it would happen before deferring

Reply:
That is what I assumed as well. Good to test it in the playground though!

Reply:
I think this makes it more confusing reading the code since we are not returning within the select/case or handling errors here.
Eg: if defer unlock was within a called function in the select/case it would be fine.

Additionally, if some code is added at the end, suddenly it would be within the lock.

Reply:
Thanks for taking a look. Agreed.